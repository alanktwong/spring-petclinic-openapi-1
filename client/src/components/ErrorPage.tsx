import * as React from 'react';

interface IErrorPageState {
  error?: {
    error: string;
    message: string;
  };
}

export default class ErrorPage extends React.Component<void, IErrorPageState> {
  constructor() {
    super();
    this.state = {};
  }

  componentDidMount() {
    fetch('http://localhost:8080/api/oops')
      .then(response => response.json())
      .then(error => this.setState({error}));
  }

  render() {
    const { error } = this.state;

    return <span>
      <img src='/images/pets.png' />

      <h2>Something happened...</h2>
      { error ?
        <span>
          <p><b>Status:</b> {error.error}</p>
          <p><b>Message:</b> {error.message}</p>
        </span>
        :
        <p><b>Unknown error</b></p>
      }
    </span>;
  }
};
