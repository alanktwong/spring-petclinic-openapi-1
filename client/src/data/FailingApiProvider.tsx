import * as React from 'react';
import { FailingApi } from 'petclinic-api';
import { Subtract } from 'utility-types';

export interface WithFailingApiProps {
  failingApi: FailingApi;
}

export const FailingApiContext = React.createContext(new FailingApi());

export const withFailingApi = <P extends WithFailingApiProps>(
  Component: React.ComponentType<P>
) =>
  class WithFailingApi extends React.Component<
    Subtract<P, WithFailingApiProps>
  > {
    static contextType = FailingApiContext;
    context!: React.ContextType<typeof FailingApiContext>;

    render() {
      return <Component failingApi={this.context} {...(this.props as P)} />;
    }
  };
