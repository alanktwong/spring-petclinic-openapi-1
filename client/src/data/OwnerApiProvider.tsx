import * as React from 'react';
import { OwnerApi } from 'petclinic-api';
import { Subtract } from 'utility-types';

export interface WithOwnerApiProps {
  ownerApi: OwnerApi;
}

export const OwnerApiContext = React.createContext(new OwnerApi());

export const withOwnerApi = <P extends WithOwnerApiProps>(
  Component: React.ComponentType<P>
) =>
  class WithOwnerApi extends React.Component<Subtract<P, WithOwnerApiProps>> {
    static contextType = OwnerApiContext;
    context!: React.ContextType<typeof OwnerApiContext>;

    render() {
      return <Component ownerApi={this.context} {...(this.props as P)} />;
    }
  };
