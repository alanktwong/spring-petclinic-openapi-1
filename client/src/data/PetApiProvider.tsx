import * as React from 'react';
import { PetApi } from 'petclinic-api';
import { Subtract } from 'utility-types';

export interface WithPetApiProps {
  petApi: PetApi;
}

export const PetApiContext = React.createContext(new PetApi());

export const withPetApi = <P extends WithPetApiProps>(
  Component: React.ComponentType<P>
) =>
  class WithPetApi extends React.Component<Subtract<P, WithPetApiProps>> {
    static contextType = PetApiContext;
    context!: React.ContextType<typeof PetApiContext>;

    render() {
      return <Component petApi={this.context} {...(this.props as P)} />;
    }
  };
