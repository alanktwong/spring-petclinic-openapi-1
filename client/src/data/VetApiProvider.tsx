import * as React from 'react';
import { VetApi } from 'petclinic-api';
import { Subtract } from 'utility-types';

export interface WithVetApiProps {
  vetApi: VetApi;
}

export const VetApiContext = React.createContext(new VetApi());

export const withVetApi = <P extends WithVetApiProps>(
  Component: React.ComponentType<P>
) =>
  class WithVetApi extends React.Component<Subtract<P, WithVetApiProps>> {
    static contextType = VetApiContext;
    context!: React.ContextType<typeof VetApiContext>;

    render() {
      return <Component vetApi={this.context} {...(this.props as P)} />;
    }
  };
