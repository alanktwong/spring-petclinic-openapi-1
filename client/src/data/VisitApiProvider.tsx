import * as React from 'react';
import { VisitApi } from 'petclinic-api';
import { Subtract } from 'utility-types';

export interface WithVisitApiProps {
  visitApi: VisitApi;
}

export const VisitApiContext = React.createContext(new VisitApi());

export const withVisitApi = <P extends WithVisitApiProps>(
  Component: React.ComponentType<P>
) =>
  class WithVisitApi extends React.Component<Subtract<P, WithVisitApiProps>> {
    static contextType = VisitApiContext;
    context!: React.ContextType<typeof VisitApiContext>;

    render() {
      return <Component visitApi={this.context} {...(this.props as P)} />;
    }
  };
