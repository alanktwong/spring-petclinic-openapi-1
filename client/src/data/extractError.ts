import { IError } from '../types';

const extractError = (response): Promise<IError> => {
  if (response.json) {
    return response.json();
  } else {
    return Promise.resolve({ message: `${response}` });
  }
};

export default extractError;
