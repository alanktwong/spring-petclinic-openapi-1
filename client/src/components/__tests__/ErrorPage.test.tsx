import React from 'react';
import renderer from 'react-test-renderer';
import ErrorPage from '../ErrorPage';
import waitForExpect from 'wait-for-expect';
import { FailingApiContext } from '../../data/FailingApiProvider';
import { FailingApi } from 'petclinic-api';

it('renders correctly', async () => {
  const mock = new FailingApi();
  mock.failingRequest = () => {
    throw {
      json: async () => ({
        message: 'error1',
      }),
    };
  };

  const component = renderer.create(
    <FailingApiContext.Provider value={mock}>
      <ErrorPage />
    </FailingApiContext.Provider>
  );

  const page = component.root.find(node => node.type.name === 'ErrorPage');

  await waitForExpect(() => {
    expect(page.instance.state.error.message).toBe('error1');
    expect(component.toJSON()).toMatchSnapshot();
  });
});
