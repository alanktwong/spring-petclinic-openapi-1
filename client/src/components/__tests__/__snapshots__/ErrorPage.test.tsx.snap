// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="alert alert-warning"
    role="alert"
  >
    <strong>
      Error!
    </strong>
     
    error1
  </div>
  <img
    alt=""
    src="/images/pets.png"
  />
</div>
`;
